### ISSUE TITLE
Issue with rectified image input

### ISSUE THREAD

**Issue by aditdoshi333** [hku-mars/livox_camera_calib#40]
Hello,

I tried with non rectified images and rectified images. Calibration output is much better with non rectified images. Is the code is expecting distorted images? Are you running rectification in the pipeline anywhere?

Regards

**Comment by ChongjianYUAN** [comment]
> Hello,
> 
> I tried with non rectified images and rectified images. Calibration output is much better with non rectified images. Is the code is expecting distorted images? Are you running rectification in the pipeline anywhere?
> 
> Regards

Yes, the input image should be distorted, and the point cloud projection needs to be provided with the same distortion model

**Comment by aditdoshi333** [comment]
Okay!

But does the final calibration matrix need to use with the rectified image or non rectified image?

**Comment by ChongjianYUAN** [comment]
> Okay!
> 
> But does the final calibration matrix need to use with the rectified image or non rectified image?

It depends on whether you correlate the LiDAR and camera data through projection, and if so, you need to use the rectified image.

**Comment by aditdoshi333** [comment]
I have one more doubt. I am able to calibrate pretty well with your code but not able to use the same matrix in r3live.  From the following can you say how is calibration?
<img width="627" alt="157250097-ab28a675-03d6-43bf-813c-7cb31d21d598" src="https://user-images.githubusercontent.com/23434856/157293391-c0d67eaf-90c1-4bb8-a249-d5bc952f0d36.png">

**Comment by ChongjianYUAN** [comment]
> I have one more doubt. I am able to calibrate pretty well with your code but not able to use the same matrix in r3live. From the following can you say how is calibration? <img alt="157250097-ab28a675-03d6-43bf-813c-7cb31d21d598" width="627" src="https://user-images.githubusercontent.com/23434856/157293391-c0d67eaf-90c1-4bb8-a249-d5bc952f0d36.png">

Sorry for the late reply. The extrinsic matrix in livox_camera_calib is from LiDAR coordinate to Camera coordinate and in R3LIVE is from Camera coordinate to LiDAR coordinate. Therefore, you should use the inverse of the output result of livox_camera_calib.
